### PR TITLE
chore: convert semanteecore into a hybrid lib+bin crate

### PR DESCRIPTION
**Pull request by mersinvald**
This feature is required for `cleanroom` (home-brew integration testing tool)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        match err.downcast::<early_exit::Error>() {
+            Ok(ee_error) => match ee_error {
+                early_exit::Error::EarlyExit(_) => (),
+            },
+            Err(other_error) => {
+                log_error_and_die!(other_error);
+            }
+        }
```
Comment (AnthonyMikh) [inline src/lib.rs]: Would be better as
```rust
if let Err(e) = err.downcast::<early_exit::Error>() {
    log::error!("{}", e);
    std::process::exit(1);
}
```

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        )
+        .build()?;
+
+    if let Err(err) = kernel.run() {
+        macro_rules! log_error_and_die {
+            ($err:expr) => {{
+                log::error!("{}", $err);
+                std::process::exit(1);
+            }};
+        }
```
Comment (AnthonyMikh) [inline src/lib.rs]: Does it make sense to define a macro and use it only once?

File: src/lib.rs (hunk 1 above)
Comment (mersinvald) [reply src/lib.rs]: That's totally different behavior. The point of this match is to not to exit with error on `early_exit` condition.

`wontfix` for now, this piece of code is a subject to change under future error-handling refactoring (https://github.com/semanteecore/semanteecore/issues/47)

File: src/lib.rs (hunk 2 above)
Comment (mersinvald) [reply src/lib.rs]: Same as the above: this is out of scope of this PR

**Pull request merged**